### ISSUE TITLE
Lagrangian trajectories in highly diffusive regions (e.g. within the mixed layer/euphotic zone) 

### ISSUE THREAD

**Issue by PaulSpence** [Midway-X/Backward_AABW#5]
My basic understanding is that lagrangian parcel trajectories work best in low diffusion regimes, i.e. in the ocean interior where the flow is adiabatic, where we can reasonably use advective model velocities to move the parcels. In higher diffusion regimes (e.g. regions of convection, water mass transformation, mixed layers) driving parcels with only model advection is problematic, because these diffusive processes play a larger role in the a "real" parcels momentum budget thate aren't represented in the model advective velocities. 

Can we get a justifiable "time in the euphotic zone" using purely advective parcel trajectories?

**Comment by PaulSpence** [comment]
This paper https://www.sciencedirect.com/science/article/pii/S1463500317301853  reviews Lagrangian particle practices Itspends a fair bit of effort discussing mixed layer/diffusion parameterizations (section 3.3.) for offline parcel trajectories.  Noting "we strongly encourage the community to gain a better understanding in how best to implement diffusion and unresolved physics for Lagrangian particles"

CMS/Parcels code offers Brownian motion for background diffusion and random displacement within the mixed layer (Table 1). But these are quite simplistic approaches for a study focussed on residence time in the mixed layer.

I am uncertain if these options will be sufficiently rigourous enough for us to justify a time in the mixed layer/euphotic zone calculation.  Unless we can clarify that the euphotic zone exists partially below the mixed layer, in a more advective layer? How well will the parcels work on the Antarctic continental shelf where AABW water mass transformation/convection takes?

**Comment by PaulSpence** [comment]
Is there an example of an offline Lagrangian parcel study that focuses on the trajectories in the mixed layer? I think most studies focus on more adiabatic regimes.

**Comment by LennartBach** [comment]
> 

-When we talk about diffusion playing a larger role, do we mean its relevance increases from 1% to 50% or from 1% to 2%?

-I would assume there is not much euphotic zone below the mixed layer (gut-feeling) so the problem is potentially significant.

-I think many lagrangian particle studies looked at trajectories in the mixed layer. For example all the plastic distribution research but also our SOBD study where particles were released in the surface.

--> We should clarify in today's meeting first if the problem is too large and we need to do something else.

**Comment by PaulSpence** [comment]
@LennartBach your avatar is a dream of mine :)

**Comment by erikvansebille** [comment]
Thanks for pointing me to this discussion, @PaulSpence! And nice to e-meet you, @LennartBach! 

Indeed, there still is need for further understanding of how to simulate particles in the mixed layer; but fortunately we've moved on a bit since the 2018 review paper 😄. 

We've especially worked quite a bit on mixing of (buoyant and non buoyant) plastic particles in the mixed layer, see for example [Onink et al (2022)](https://doi.org/10.5194/gmd-15-1995-2022) and [Fischer et al (2022)](https://doi.org/10.5194/bg-19-2211-2022). I think that these methods could be extended also for purely passive water parcels?

By the way, we are also working on tracking Dissolved Inorganic Carbon (in the North Atlantic) with Parcels, in @daanreijnders' PhD project. So may be good to make sure we're not duplicating efforts!

**Comment by StephenGriffies** [comment]
I agree with @erikvansebille and @PaulSpence : treating Lagrangian particles in the boundary layer, using a hydrostatic model, is tough.  Erik mentions some progress on parameterizing the motion for plastic particles.  That could presumably be extended to passive fluid particles.  But it will be a parameterization, with the attendant uncertainties.  In effect, you are looking for a boundary layer parameterization...that is of course a full problem in itself, and something perhaps not in your direct interests.  

@LennartBach asked is this a 1-2% change or 1-50% change?  The answer depends on the model flow field that is resolved for the particles.  Say you have a 1km hydrostatic model with dz=2m in the boundary layer, and you save (u,v,w) very time step (or perhaps every hour).  The particles would move around a lot in the vertical and the trajectories would presumably look very similar to the same model run in nonhydrostatic mode.  But for a 10km model with coarser dz that is sampled every day (at best), then the story gets more confused since so much of the submesoscale and non-hydrostatic processes are now missing from the resolved velocity field.  

Perhaps the safest approach is to use particles in combination with a suite of ventilation tracers.  Tracers feel the vertical diffusive effects that Lagrangian particles will only feel indirectly, but they have limited flexibility since they generally need to be run online. 

Do you have the option of rerunning the model with new/more dye tracers?  You could perhaps run some tests with high sampling (say hourly 3d velocity) and comparing to tracers.  

Bottomline: this project will require some careful work on the method in order for readers to consider your results robust.

**Comment by erikvansebille** [comment]
Thanks @StephenGriffies for these comments; I fully agree! I know that @nordam has also done some work on comparison between Eulerian and Lagrangian methods for vertical mixing. Don't think it's submitted yet, but perhaps he has some bits to share already?

**Comment by nordam** [comment]
Hello!

I'm more of a numerical person than an oceanographer, so the relevance of adiabatic flow and the euphotic zone and so on is not 100% clear to me.

From a purely numerical point of view, I would say that in principle it's possible to get the same results from an offline Lagrangian particle model as you would get from an online tracer calculation inside the ocean model, provided you use the same velocities for advection, the same diffusivities, provide output sufficiently often, and use a short enough timestep and a consistent scheme for evaluating the diffusivity and it's derivatives. But I'm not sure if that was the question?

**Comment by nordam** [comment]
To expand a bit on my comment above (still from a numerical point of view):

Whether random displacement is a "simplistic approach" depends on what you want to model. A random displacement scheme modelled by the stochastic differential equation

dz = (v(z,t) + K'(z,t)) dt + sqrt(2K(z,t)) dW

_is_ mathematically equivalent to the advection-diffusion equation with advection v and diffusivity K, provided that v(z, t), K(z, t), and K'(z,t) are sufficiently smooth functions. (I believe the formal requirement is that they should all be Lipshitz continuous in z). For some more details, see for example Appendix A in this paper: https://www.sciencedirect.com/science/article/pii/S0025326X19305338 

(I apologise for promoting my own work, but I'm not aware of any other source that writes this down specifically for the advection-diffusion equation in a reasonably compact form. For more generality and mathematical detail, see for example page 37 or so of Kloeden & Platen (https://link.springer.com/book/10.1007/978-3-662-12616-5) or pages 96-102 in Gihman & Skorokhod (https://link.springer.com/book/9783642882661).)

Also, note that while I write this down for one dimension here, the same holds in 3D, see for example this paper by Spivakovskaya et al, which deals with the general case of a diffusion tensor with off-diagonal elements: https://link.springer.com/article/10.1007/s10236-007-0102-9

And finally, note that when I say the random displacement scheme is mathematically equivalent to the advection-diffusion scheme, what I mean is that the probability distribution of particle positions will evolve in the same way as a concentration modelled by the advection-diffusion equation. Solving for a sufficiently large number of particles will allow you to approximate the probability distribution, and thus the concentration.



So, the point is that random displacement is not a simplistic approach if you want to have a Lagrangian model for advection and diffusion, which is equivalent to the advection-diffusion equation. It is in fact the correct approach, provided you can meet the required conditions. And in practice, those conditions may not have to be met exactly. For example you can often get away with a non-continuous z-derivative of K(z, t), as long as your timestep is short enough. And you might even get away with a non-continuous K(z, t) if you use a suitable numerical scheme: https://os.copernicus.org/articles/3/525/2007/

Regarding the specific question about modelling residence time in the euphotic zone, I would say that this depends not so much on the Lagrangian approach, but rather on the quality of the data you use to force the model. The Lagrangian model doesn't know about euphotic zones and adiabatic flows and so on, it only asks about the advection velocity, the diffusivity, and usually (depending on your numerical scheme) the derivative of the diffusivity. If you have those variables available with sufficiently high resolution to capture the features that are relevant for the particle transport, then the Lagrangian scheme should be able to model the transport correctly.

Of course, the stochastic differential equation describing the random displacement scheme is one thing. Actually solving it numerically is something else. Different schemes have different strengths and weaknesses, and may be more or less suitable for different situations. A relevant example may be this paper by Gräwe, for a discussion of how different schemes perform for modelling residence time in the surface mixed layer, above a sharp pycnocline: https://link.springer.com/article/10.1007/s10236-012-0523-y

Feel free to let me know if I'm missing the point, or if the explanation doesn't make sense.

**Comment by PaulSpence** [comment]
@nordam @StephenGriffies @erikvansebille Thank you all for the sage advice :) Your time and wisdom are greatly appreciated. @Midway-X is conducting some comparisons of DSW seeded parcels to existing online AABW tracer dye simulations: https://doi.org/10.1029/2021GL097211. If the comparisons look okay, maybe we can come up with a range of residence time estimates using different diffusion schemes. However, I think the current dt=1day offline velocities maybe a strong limitation.

**Comment by LennartBach** [comment]
Hi @erikvansebille @StephenGriffies @nordam, it is great to meet you all. Thanks a lot for taking your time and sharing your insights. This is hugely helpful for us. I try to summarise what I see as the 'to do list' from your comments. I am a biological oceanographer by training and therefore apologise for silly questions/conclusions in advance.

Key take home from your comments is that Lagrangian particle tracking is potentially useful for our scientific question but we need to do some testing to first verify its usefulness. To do this we do the following
--> Read the papers you suggested (thanks a lot for providing these!)
--> We will seed particles at a location on the Antarctic Shelf where Andy Hogg has added an online tracer in the 1/10° Access model (for ventilation I believe) and compare the differences between online tracer and virtual particles. This is what @StephenGriffies also suggested and I am glad you made this suggestion as your idea is consistent with what we had discussed to do within our last meeting with @Midway-X.
--> We'll probably use the highest temporal/spatial resolution available for the particle simulation.

Luckily, @Midway-X uses parcels and not CMS any longer as there is so much good development going into parcels.

**Comment by StephenGriffies** [comment]
I agree with @nordam.  The key point is that one needs to save the 3d diffusivity and its derivative at a high time sampling, along with (u,v,w).  As the boundary layer diffusivity is highly transient, it will need to be sampled at a high frequency. Alas, I doubt the diffusivity is available for @LennartBach at the needed frequency, which then makes me concerned that there will be differences between the Lagrangian particles and tracers.  But that concern can be ameliorated with careful tests.   

@LennartBach , your tests seem reasonable.  Having been on both sides of the author/review process with Lagrangian studies, it is very useful to have evidence to quantify the limitations of your method. Doing so provides some basis for error bars.  Sometimes the best method is not the most accurate method, but instead it is the most flexible method whose limitations/error bars are well quantified.
